File: csharp/HtmlReportTest.cs
using System.Collections.Generic;
using System.IO;
using NUnit.Framework;

namespace ReportGeneratorCS.Tests
{
    [TestFixture]
    public class HtmlReportTest {

        [Test]
        public void testReport() {
            IList<Machine> line = new List<Machine>();
            line.Add(new Machine("mixer", "left"));

            Machine extruder = new Machine("extruder", "center");
            extruder.Put("paste");
            line.Add(extruder);

            Machine oven = new Machine("oven", "right");
            oven.Put("chips");
            line.Add(oven);

            Robot robot = new Robot();
            robot.MoveTo(extruder);
            robot.Pick();

            StringWriter output = new StringWriter();
            HtmlReport.report(output, line, robot);

            string expected = "<h1>FACTORY REPORT</h1>\n"
                              + "<h2>mixer</h2>\n"
                              + "<ul>\n"
                              + "<li>location = left</li>\n"
                              + "<li>no bin</li>\n"
                              + "</ul>\n"
                              + "<h2>extruder</h2>\n"
                              + "<ul>\n"
                              + "<li>location = center</li>\n"
                              + "<li>no bin</li>\n"
                              + "</ul>\n"
                              + "<h2>oven</h2>\n"
                              + "<ul>\n"
                              + "<li>location = right</li>\n"
                              + "<li>bin containing chips</li>\n"
                              + "</ul>\n"
                              + "<h2>Robot</h2>\n"
                              + "<ul>\n"
                              + "<li>location = extruder</li>\n"
                              + "<li>bin containing paste</li>\n"
                              + "</ul>\n";

            Assert.That(expected, Is.EqualTo(output.ToString()));
        }
    }
}

File: java/HtmlReportTest.java
import junit.framework.TestCase;

import java.util.ArrayList;
import java.io.StringWriter;
import java.io.IOException;

public class HtmlReportTest extends TestCase {
	public HtmlReportTest(String name) {
		super(name);
	}

	public void testReport() throws IOException {
		ArrayList<Machine> line = new ArrayList<Machine>();
		line.add(new Machine("mixer", "left"));

		Machine extruder = new Machine("extruder", "center");
		extruder.put("paste");
		line.add(extruder);

		Machine oven = new Machine("oven", "right");
		oven.put("chips");
		line.add(oven);

		Robot robot = new Robot();
		robot.moveTo(extruder);
		robot.pick();

		StringWriter out = new StringWriter();
		HtmlReport.report(out, line, robot);

		String expected = "<h1>FACTORY REPORT</h1>\n"
				+ "<ul>\n<li>Machine mixer</li>\n<li>Machine extruder</li>\n"
				+ "<li>Machine oven bin=chips</li>\n</ul>\n"
				+ "<p>Robot location=extruder bin=paste</p>\n" + "<hr>\n";

		assertEquals(expected, out.toString());
	}
}
